src/javascripts/ng-admin/Crud/form/FormController.js

```diff
@@ -71,20 +72,18 @@ export default class FormController {
             .then(() => $translate('CREATION_SUCCESS'))
             .then(text => notification.log(text, { addnCls: 'humane-flatty-success' }))
             .catch(error => {
-                const errorMessage = this.config.getErrorMessageFor(this.view, error) || 'ERROR_MESSAGE';
+                // this is kept for backward compatibility with the entity.errorMessage() method 
+                const msg = this.config.getErrorMessageFor(this.view, error) || 'ERROR_MESSAGE';
+                error.data.message = msg;
+                
+                const errorMessage = this.HttpErrorService.handleError(error);
```

Comment:
@jpetitcolas, I tried to write that damn test case but its proving to be quiet a pain. The reason is that  FormController is rather a big class, and I want to only test a very small part of it.
Since I dont have big experience with Karma, I may going at this the wrong way. I have mocked quiet a few services, and tried to use .and.callThrough(), but when invoking writeQueries it is all falling apart :(
Would you mind having a quick look to see where I am going wrong ? Thanks !


```
const  FormController = require('../../../js/core/controllers/FormController.js');

describe('To Test FormController', function(){
    let mockHttpErrorService;
    let mockFormController;
    let view;
    let $state;
    let $injector;
    let $translate;
    let previousState;
    let WriteQueries;
    let Configuration;
    let progression;
    let notification;
    let dataStore;
    let $scope;
    let $event;
    let $rootScope;

    angular.module('testFormController', [])
    .service('Configuration',() => ({
    }))
    .service('view', function() {
        return {
            title: jasmine.createSpy('view.title'),
            description: jasmine.createSpy('view.description'),
            actions: jasmine.createSpy('view.actions'),
            fields: jasmine.createSpy('view.fields'),
            getEntity: function(){
                return {
                    uniqueId:"",
                    identifier:function(){
                        return{
                            name:function(){return "val"}//jasmine.createSpy('view.getEntity.identifier.name')
                        }
                     }
                }
                } 
        };
    })
    .service('$state', function() {
        return {
        };
    })
    .service('$event', function() {
        return {
            preventDefault:jasmine.createSpy('$event.preventDefault')
        };
    })
    .service('dataStore', function() {
        return {
            getFirstEntry:function(id){
                return{
                    values:{}
                }
            }
        };
    })
    .service('$translate', function() {
        return {
        };
    })
    .service('previousState', function() {
        return {
        };
    })
    .service('WriteQueries', function() {
        return {
            
            createOne:jasmine.createSpy('WriteQueries.createOne')
       };
    })
    .service('progression', function() {
        return {
       };
    })
    .service('notification', function() {
        return {
        };
    })
    .service('HttpErrorService', function() {
        return {
            handleError: jasmine.createSpy('HttpErrorService.handleError')
        };
    })
  .controller('FormController', ['$scope','$state','$injector', '$translate', 'previousState', 'WriteQueries', 'Configuration', 'progression', 'notification','view','dataStore','HttpErrorService',FormController]);

    beforeEach(angular.mock.module('testFormController'));

    beforeEach(inject(function(_$event_,_HttpErrorService_, $controller, _view_,_$state_, $injector, _$translate_, _previousState_, _WriteQueries_, _Configuration_, _progression_, _notification_,_dataStore_) {
        Configuration= _Configuration_;
       $rootScope = $injector.get('$rootScope');
        $scope = $rootScope.$new();
        view = _view_;
        $state= _$state_;
        dataStore=_dataStore_;
        $translate= _$translate_;
        previousState= _previousState_;
        WriteQueries= _WriteQueries_;
        progression= _progression_;
        notification= _notification_;
        $event = _$event_;
        mockHttpErrorService = _HttpErrorService_;
        mockFormController = $controller('FormController', {$scope:$scope,$state:$state, $injector:"", $translate:$translate, previousState:previousState, WriteQueries:WriteQueries, Configuration:function conf(){}, progression:progression, notification:notification,view:view, dataStore:dataStore,HttpErrorService:mockHttpErrorService});
    }));

    it('should be able to call HttpErrorService', () => {
        var error = {status:403}
        spyOn(mockFormController, 'validateEntry')
                spyOn(mockFormController, 'WriteQueries')

        mockFormController.submitCreation($event);
        try {
            expect(WriteQueries.createOne().toThrow(new Error("403")));
       } catch (e) {
       }
        
    });

});

```